tests/editor/test_editor_translation.h

```diff
@@ -0,0 +1,94 @@
+/**************************************************************************/
+/*  test_editor_translation.h                                             */
```

Comment:
I tried to add unit tests. The examples for unit tests which affect the editor are sparse, so I might have gotten something wrong.

These tests could exist in `editor/tests/` rather than `tests/editor/`, like the tests for gdscript. It seems like there is very little precedent here for editor unit tests.